zookeeper-server/src/main/java/org/apache/zookeeper/server/Request.java

```diff
@@ -462,4 +476,48 @@ public String getUsers() {
         }
         return users.toString();
     }
+
+    /**
+     * This will check if the request's handler supports skip feature,
+     * the information that is passed from Learner to LearnerMaster using protocol version info
+     *
+     * See org.apache.zookeeper.server.quorum.QuorumZooKeeperServer.PROTOCOL_VERSION
+     *
+     */
+    public boolean canSkip() {
+        return skipRequestHandler != null;
```

Comment:
Wouldn't this always return true? As far as I can see skipRequestHandler will never be null as it's set by `setSkipRequestHandler`

Reply:
True, this is related to learner protocol version, something outside the scope of this. This is not required to be in this pull request.